### DISCUSSION TITLE
Pld Optimized Rotation Change

### DISCUSSION THREAD

**Discussion by MarvinAkil** [daemitus/XIVComboPlugin#418] [1 - Tanks]
At the moment the mod uses atonement -> fast blade -> riot blade -> supplication -> sepulchre -> Holy Spirit.

Optimal filler rotation is atonement -> fast blade -> riot blade -> supplication -> Holy Spirit -> sepulchre 

Could this be changed into that rotation. Cause atm its not useful to use it at all as it is not optimal at all

**Comment by MKhayle** [comment]
@kaedys hi, have fun

**Comment by kaedys** [comment]
You put it incredibly confusingly, tbh, but suffice it to say, the order specifically of Holy Spirit and _Supplication_ doesn't matter.  They have identical potency.  Technically there's a tiny chance that you'd have to disengage during that GCD of FoF and thus Holy Spirit could be used where Supplication could not, but if you're worried about optimization on that scale, you shouldn't be using a mod like this in the first place (or at the least, not using the Atonement Royal Authority feature).

I can swap the priority of Holy Spirit and Supplication/Sepulchre, though.  That's pretty easy.  Worth noting that this is a pretty niche case (though to be fair, the option in question _does_ specify it is an advanced optimization option, so that tracks).  The _only_ time this can make a difference is if FoF comes off cooldown (or gets used) precisely during the 2nd GCD before Royal Authority, so whatever is left unused before that Royal Authority is the first of the 3 spare GCDs in FoF.  In that case, it's a gain of 40p to be able to use Sepulchre -> Royal Authority -> Holy Spirit over the current Holy Spirit -> Royal Authority -> Holy Spirit that would be used.

**Comment by MarvinAkil** [comment]
Im just writing this, as the combo i have written above is the one which is specified by balance. It does matter as it will make Sure to delay sepulchre as much as you can. Atm sepulchre is used 1 gcd earlier than it should be. 

Its not a necessary change if you dont want to. Its definitly the filler which guarantees as much sepulchre in your fof as you want 

![Screenshot_20240811_194153_Discord](https://github.com/user-attachments/assets/a7a7f679-7d52-4cdb-b3cb-5678cd445d4e)


![image](https://github.com/user-attachments/assets/965e5a0b-2e40-458e-9d69-9d0340b4f198)

**Comment by kaedys** [comment]
I'm swapping the priority of Holy Spirit and finishing the Atonement combo.  That will place Sepulchre in the correct spot, used _immediately_ prior to Royal Authority.  That part wasn't an issue.

I was planning on _just_ swapping the priority of Holy Spirit and finishing the Atonement combo, which would put Sepulchre in the correct spot but swap Supplication and Holy Spirit (so the "normal" combo outside FoF would have been Holy Spirit -> Supplication -> Sepulchre instead of Supplication -> Holy Spirit -> Sepulchre).  However, the code to handle the actual recommended order isn't as bad as I thought, and looking at it, it's actually two GCDs in FoF it provides disconnect protection against, rather than just one GCD, _and_ provides some minor disconnect protection for a further GCD during the normal combo, so I'm going to do that part as well.

**Comment by MarvinAkil** [comment]
Perfect!

**Comment by kaedys** [comment]
Ok, well, I now have _forty lines_ of comments in the Paladin file, in 3 locations within the main combo block, just annotating all the wacky shit we're doing to optimize Atonement and Holy Spirit in and out of FoF. O.O

**Reply by MarvinAkil** [reply]
![image](https://github.com/user-attachments/assets/dc9f1f27-1960-4684-b97d-753df8d9583c)

**Reply by kaedys** [reply]
I'm definitely in camp "it's better to have too much documentation than not enough" >.>

**Reply by MKhayle** [reply]
> Ok, well, I now have _forty lines_ of comments in the Paladin file, in 3 locations within the main combo block, just annotating all the wacky shit we're doing to optimize Atonement and Holy Spirit in and out of FoF. O.O

and despite those comments I was unable to figure out whether the Holy Spirit/Holy Combo issue was a secret nerd balance optimization or a c/p error 😆

**Reply by kaedys** [reply]
Haha, that's partially because I didn't really annotate the Prominence combo at all, because it doesn't really _have_ any optimization.  The only change that happens over there is that Holy Circle is only used when it would be overwritten, except during FoF where it is used immediately.

**Reply by MarvinAkil** [reply]
> I'm definitely in camp "it's better to have too much documentation than not enough" >.>

definetly better to be in that camp

**Comment by kaedys** [comment]
Since it didn't auto-link here, #419
